bindings/rust/src/lib.rs

```diff
@@ -133,6 +133,75 @@ impl From<f64> for Value {
     }
 }
 
+/// A WebAssembly value i32/i64/f32/f64 with its type specified.
+pub enum TypedValue {
+    I32(i32),
+    U32(u32),
+    I64(i64),
+    U64(u64),
+    F32(f32),
+    F64(f64),
+    Void,
```

Comment:
I think we don't need Void as we should use `Option<>` instead.

Reply:
I don't know Rust style, but seems Void will be useful in C/C++ in some cases.

Reply:
Using Result/Option is more natural than having a special enum value in this case, it seems.